iroh-embed/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+use anyhow::Result;
+use iroh_api::Api;
+use iroh_rpc_client::Config as RpcClientConfig;
+
+/// An address allowing internal communication with the iroh store service.
+///
+/// This is only needed to wire up the several services together and should not be used to
+/// communicate directly to the service.
+///
+/// Can be created from [`store::RocksStoreService::addr`].
+pub use iroh_rpc_types::p2p::P2pClientAddr;
+
+/// An address allowing internal communication with the iroh p2p service.
+///
+/// This is only needed to wire up the several services together and should not be used to
+/// communicate directly to the service.
+///
+/// Can be created from [`p2p::P2pService::addr`].
+pub use iroh_rpc_types::store::StoreClientAddr;
+
+mod p2p;
+mod store;
+
+pub use p2p::P2pService;
+pub use store::RocksStoreService;
+
+/// The full iroh system.
+///
+/// Creating this will create an iroh system and start several tokio tasks.  To make the
+/// system do anything use the [`Iroh::api`] function to get an API.
+#[derive(Debug)]
+pub struct Iroh {
+    store: RocksStoreService,
+    p2p: P2pService,
+    api: Api,
+}
+
+impl Iroh {
+    /// Creates a new running iroh system.
+    ///
+    /// To create an iroh system first a store and p2p service must be created and must be
+    /// wired up to communicate together.  This means the [`P2pService`] must be created
+    /// with the [`StoreClientAddr`] from the store used.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// tokio_test::block_on(async {
```

Comment:
```suggestion
    /// # tokio_test::block_on(async {
```

Reply:
not entirely sure what is going on, but i suspect this is a review of an earlier commit?  but github doesn't seem to say so?